src/dotty/tools/dotc/parsing/Parsers.scala

```diff
@@ -648,12 +648,18 @@ object Parsers {
     }
 
 /* ------------- TYPES ------------------------------------------------------ */
-    /** Same as [[typ]], but emits a syntax error if it returns a wildcard.
+    /** Same as [[typ]], but if this results in a wildcard it emits a syntax error and
+     *  returns a tree for type `Any` instead.
      */
     def toplevelTyp(): Tree = {
       val t = typ()
-      for (wildcardPos <- findWildcardType(t)) syntaxError("unbound wildcard type", wildcardPos)
-      t
+      findWildcardType(t) match {
+        case Some(wildcardPos) =>
+          syntaxError("unbound wildcard type", wildcardPos)
+          Select(Select(Ident(Names.termName("_root_")), Names.termName("scala")),
```

Comment:
@cswinter Jave a look at the definition of `scalaUnit` in `untpd`. I propose to model a `scalaAny` like this instead of writing things out in the parser.